Plug-Ins/NovaTools/System Editor/StellarView.swift

```diff
@@ -0,0 +1,210 @@
+import AppKit
+import RFSupport
+
+class StellarView: NSView {
+    let resource: Resource
+    let manager: RFEditorManager
+    let isEnabled: Bool
+    private(set) var position = NSPoint.zero
+    private(set) var image: NSImage? = nil
+    private var attributes: [NSAttributedString.Key: Any] {
+        [.foregroundColor: NSColor.white, .font: NSFont.systemFont(ofSize: 11)]
+    }
+    private var systemView: SystemMapView? {
+        superview as? SystemMapView
+    }
+    var point = NSPoint.zero {
+        didSet {
+            frame.origin.x = point.x - frame.size.width / 2
+            frame.origin.y = point.y - frame.size.height / 2
+        }
+    }
+    var isHighlighted = false {
+        didSet {
+            if oldValue != isHighlighted {
+                needsDisplay = true
+            }
+        }
+    }
+    var highlightCount = 1 {
+        didSet {
+            if oldValue != highlightCount {
+                needsDisplay = true
+            }
+        }
+    }
+
+    init?(_ stellar: Resource, manager: RFEditorManager, isEnabled: Bool) {
+        self.resource = stellar
+        self.manager = manager
+        self.isEnabled = isEnabled
+        super.init(frame: .zero)
+        do {
+            try self.read()
+        } catch {
+            return nil
+        }
+    }
+
+    func read() throws {
+        let reader = BinaryDataReader(resource.data)
+        position.x = Double(try reader.read() as Int16)
+        position.y = Double(try reader.read() as Int16)
+
+        let spinId = (Int)(try reader.read() as Int16) + 1000;
```

Comment:
It's a weird quirk of Swift that allows the parentheses to work like that. They're redundant and I prefer to avoid the appearance of a C-style cast.
Also please avoid semi-colons 😄 
```suggestion
        let spinId = Int(try reader.read() as Int16) + 1000
```

Reply:
Argh after reflexively adding them almost every line and then removing I let one through after all!